src/pages/Courses/index.tsx

```diff
@@ -1,21 +1,125 @@
 import React from "react";
 import CourseCard from "../../components/CourseCard/index";
-import { withStyles } from "@material-ui/core";
-import { styles } from "./style";
-import { Props } from "./types";
+import {withStyles} from "@material-ui/core";
+import {styles} from "./style";
+import {Props, State} from "./types";
 import {data} from '../../data'
+import {Course} from "./types"
+import Modal from "../../components/Modal";
+import Typography from '@mui/material/Typography';
+import Fab from "@mui/material/Fab";
+import AddIcon from "@mui/icons-material/Add";
+import Button from '@mui/material/Button';
+
+
+class CoursePage extends React.Component<Props, State> {
+    maxId = 100;
+    state = {
+        data,
+        selectedCourse: null,
+        open: false,
+        inputValues: { img: '', name: '', course: '', author: '', stars: '', price: ''},
+
+    };
+
+    onSubmitHandler = () => {
+
+        this.setState(({data}) => ({
+            data: [...data,
+                {
+                    id: Symbol('id'),
+                    info: '',
+                    img: this.state.inputValues.img,
+                    name: this.state.inputValues.name,
+                    course: this.state.inputValues.course,
+                    author: this.state.inputValues.author,
+                    stars: Number(`${this.state.inputValues.stars}`),
+                    price: Number(`${this.state.inputValues.price}`),
+                }]
+        }))
+    }
+
+    //     this.setState(({ data }) => ({ data: [...data, course] }))
+// input value={this.state.name} onChange={(value) => this.setState({ name: value })}
+
+    handleDelete = (id: number) => () => {
+        this.setState(({data}) => ({
+            data: data.filter((item) => item.id !== id)
+        }));
+    };
+
+    handleSelectCourse = (course: Course) => () => {
+        this.setState({selectedCourse: course})
+    }
+
+    handleCloseDetailsModal = () => {
+        this.setState({selectedCourse: null})
+    }
+    handleOpenModal = () => {
```

Comment:
You can combine two functions `handleOpenModal` and `handleCloseModal` into `handleToggleModal`
`setState(({ open }) => ({ open: !open }))`